htgotts.go

```diff
@@ -68,46 +73,51 @@ func (speech *Speech) Speak(text string) error {
 	return speech.PlaySpeechFile(fileName)
 }
 
-/**
- * Create the folder if does not exists.
- */
+// Read downloads speech and plays it without writing to disk
+func (speech *Speech) Read(text string) error {
+	buf, err := speech.fetch(text)
+	if err != nil {
+		return err
+	}
+	return speech.Handler.PlayBuf(buf)
+}
+
+// Create the folder if does not exists.
 func (speech *Speech) createFolderIfNotExists(folder string) error {
 	dir, err := os.Open(folder)
 	if os.IsNotExist(err) {
-		return os.MkdirAll(folder, 0700)
+		return os.MkdirAll(folder, 0o700)
 	}
 
 	dir.Close()
 	return nil
 }
 
-/**
- * Download the voice file if does not exists.
- */
+// Download the voice file if does not exists.
 func (speech *Speech) downloadIfNotExists(fileName string, text string) error {
 	f, err := os.Open(fileName)
 	if err != nil {
-		url := fmt.Sprintf("http://translate.google.com/translate_tts?ie=UTF-8&total=1&idx=0&textlen=32&client=tw-ob&q=%s&tl=%s", url.QueryEscape(text), speech.Language)
-		response, err := http.Get(url)
-		if err != nil {
-			return err
-		}
-		defer response.Body.Close()
-
-		output, err := os.Create(fileName)
-		if err != nil {
-			return err
+		if os.IsNotExist(err) {
+			buf, err := speech.fetch(text)
+			if err != nil {
+				return err
+			}
+
+			output, err := os.Create(fileName)
+			if err != nil {
+				return err
+			}
+
+			_, err = io.Copy(output, buf)
 		}
-
-		_, err = io.Copy(output, response.Body)
 		return err
 	}
 
 	f.Close()
 	return nil
 }
 
-func (speech *Speech) generateHashName(name string) string {
-	hash := md5.Sum([]byte(name))
+func (speech *Speech) fnameFromHash(text string) string {
```

Comment:
Please, if possible, do not use short names for the readable code. My suggestion for example `fileNameFromHash`